src/gui/plugins/component_inspector/ComponentInspector.qml

```diff
@@ -216,6 +216,26 @@ Rectangle {
         }
       }
 
+      ToolButton {
+        id: addSystemButton
+        checkable: false
+        text: "\u002B"
+        contentItem: Text {
+          text: addSystemButton.text
+          color: "#b5b5b5"
+          horizontalAlignment: Text.AlignHCenter
+          verticalAlignment: Text.AlignVCenter
+        }
+        visible: entityType == "model"
```

Comment:
It should be ok to add systems to models, the world, visuals and sensors.

Reply:
updated check. [da4f16a](https://github.com/gazebosim/gz-sim/pull/1549/commits/da4f16a0067dcce3bd0ae96d543786058646effc)

Reply:
Yeah that `material` type call looks mistaken.

How about `world`, does that work if we add it?

Reply:
Humm I tried adding `world`, but then when I add systems to the world they don't work properly. I tried adding `Buoyancy` to `shapes.sdf`, and also removed `Physics` from `rolling_shapes` and tried adding it from the GUI.

Reply:
added support for world. [d7503c5](https://github.com/gazebosim/gz-sim/pull/1549/commits/d7503c5b099a57919774a329f1533423110fde45). I tested adding `Buoyancy` to `shapes.sdf` and I I think it's working for me. I see it appear in the `System Plugin Info` list in the Component Inspector, and when I add a new object into the world, it immediately shoots upwards.

Reply:
This is what I did:
![add_system_gui_shapes](https://user-images.githubusercontent.com/4000684/175751678-b6cab0e3-8119-4d25-91d3-5e1562e6b193.gif)

Reply:
>  when I add a new object into the world, it immediately shoots upwards.

How about the existing objects, they should also shoot upwards, right? It looks like the new world plugins aren't acting on pre-existing entities. It's the same issue as adding physics to `rolling_shapes`.

Reply:
> How about the existing objects, they should also shoot upwards, right?

I think that depends on how the system is written? I see that in the Buoyancy plugin for example, it uses `EachNew` to do some preprocessing. ?However, when a system is inserted at runtime, that logic will not be executed for existing objects.

Reply:
Ahh makes sense